src/main/java/com/cleanroommc/groovyscript/compat/mods/theaurorian/MoonlightForge.java

```diff
@@ -0,0 +1,104 @@
+package com.cleanroommc.groovyscript.compat.mods.theaurorian;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.shiroroku.theaurorian.Recipes.MoonlightForgeRecipe;
+import com.shiroroku.theaurorian.Recipes.MoonlightForgeRecipeHandler;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class MoonlightForge extends VirtualizedRegistry<MoonlightForgeRecipe> {
+
+    @RecipeBuilderDescription(example = @Example(".input(item('minecraft:stone_sword'), item('minecraft:diamond')).output(item('minecraft:diamond_sword'))"))
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        restoreFromBackup().forEach(MoonlightForgeRecipeHandler::addRecipe);
+        removeScripted().forEach(r -> MoonlightForgeRecipeHandler.allRecipes.removeIf(u -> u.equals(r)));
+    }
+
+    public void add(MoonlightForgeRecipe recipe) {
+        addScripted(recipe);
+        MoonlightForgeRecipeHandler.addRecipe(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('theaurorian:aurorianiteingot')"))
+    public boolean removeByInput(IIngredient output) {
+        return MoonlightForgeRecipeHandler.allRecipes.removeIf(r -> {
+            if (output.test(r.getInput1()) || output.test(r.getInput2())) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example("item('theaurorian:queenschipper')"))
+    public boolean removeByOutput(IIngredient output) {
+        return MoonlightForgeRecipeHandler.allRecipes.removeIf(r -> {
+            if (output.test(r.getOutput())) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        MoonlightForgeRecipeHandler.allRecipes.forEach(this::addBackup);
+        MoonlightForgeRecipeHandler.allRecipes.clear();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<MoonlightForgeRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(MoonlightForgeRecipeHandler.allRecipes).setRemover(r ->
+            MoonlightForgeRecipeHandler.allRecipes.removeIf(u -> {

```

Comment:
could this code be moved to a `remove` function?

Reply:
I suppose `removeByInput`/`removeByOutput` don't have to be modified to use that method? Seems like either an unnecessary performance bottleneck or an unnecessary bunch of type casts

Reply:
nope, just prefer having a basic `public boolean remove(T)` method for `streamRecipes` to work with, and thats whats done in most registries using `streamRecipes`

Reply:
should work